src/cart/index.jsx

```diff
@@ -1,7 +1,90 @@
 import React, { Component } from 'react'
+import { numberWithCommas } from '../functions/index'
+import { CartItem } from './components/CartItem'
 
 export default class Cart extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      cart: []
+    }
+  }
+
+  componentDidMount() {
+    this.setState({
+      cart: this.props.cart
+    })
+  }
+
+  setQuantity = e => {
+    const { cart } = this.state
+    const id = Number(e.target.id)
+    const quantity = Number(e.target.value)
+    this.setState({
+      cart: cart.map(item => {
+        return item.id === id ? { ...item, quantity: quantity } : item
+      })
+    })
+    this.props.updateProductToCart(
+      cart.map(item => {
```

Comment:
I think its a bit redundant here. We can refactor this method using `setState` callback.

```jsx
this.setState({
  cart: cart.map....
}, () => {
  this.props.updateProductToCart(this.state.cart)
})
```